src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs

```diff
@@ -190,7 +190,7 @@ internal static void ValidateKeyValuePair(string keyword, string value)
             }
             if ((null != value) && value.Contains('\0'))
             {
-                throw ADP.InvalidValue(keyword);
+                throw ADP.InvalidValue(value);
```

Comment:
Actually, why does this need to change to value? The argument here is the parameter name, right?

Reply:
I am actually considering that it should be `nameof(value)`. It's expecting the parameter name, returning an `ArgumentException`.

https://source.dot.net/#System.Data.Common/System/Data/Common/AdapterUtil.Common.cs,723f78a5c15d5418,references

Reply:
Unless you mean that the value of keyword should be the parameter name since it's triggered when setting the indexer and it is meant to show this as the source

https://source.dot.net/#System.Data.Common/System/Data/Common/DbConnectionStringBuilder.cs,98

Reply:
> Unless you mean that the value of keyword should be the parameter name since it's triggered when setting the indexer and it is meant to show this as the source
> 
> https://source.dot.net/#System.Data.Common/System/Data/Common/DbConnectionStringBuilder.cs,98

Right, I suspect that was the original intent.